src/components/Modal/Modal.tsx

```diff
@@ -0,0 +1,42 @@
+import React from "react";
+
+import * as Dialog from "@radix-ui/react-dialog";
+import {
+  ContentWrapper,
+  StyledClose,
+  StyledDescription,
+  StyledOverlay,
+  StyledTitle,
+} from "./Modal.styled";
+
+interface ModalProps
+  extends Omit<React.ComponentProps<typeof Dialog.Root>, "children"> {
+  width?: string;
+  children: (props: {
+    Title: typeof StyledTitle;
+    Description: typeof StyledDescription;
+    Close: typeof StyledClose;
+  }) => React.ReactNode;
+}
+
+export function Modal({ children, width, ...rootProps }: ModalProps) {
```

Comment:
Should the modal deal with key events like <kbd>escape</kbd>? Maybe `@radix-ui/react-dialog` already has an option to enable that?